go/client/chat_cli_alias.go

```diff
@@ -0,0 +1,97 @@
+package client
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/keybase/client/go/protocol/chat1"
+)
+
+// FullConversationAlias is the full alias of a conversation ID. It is the ":"
+// plus lower case hex string of the conversation ID, with first 2 bytes (4 hex
+// characters) moved to the end. This is because first 2 bytes are for shard
+// IDs, and for now is always 0.
+type FullConversationAlias string
+
+// ShortConversationAlias is a prefix of a FullConversationAlias.
+type ShortConversationAlias string
+
+func ConversationAliasWidth(numOfBytes int) int {
+	// ":" + hex encoding of bytes
+	return 1 + numOfBytes*2
+}
+
+func MakeFullConversationAlias(cid chat1.ConversationID) FullConversationAlias {
+	cidHex := strings.ToLower(cid.String())
+	return FullConversationAlias(":" + cidHex[4:] + cidHex[:4])
+}
+
+func (f FullConversationAlias) mustBeValid() {
+	if len(f)%2 != 1 || f[0] != ':' {
+		panic(fmt.Sprintf("incorrect format of FullConversationAlias: %s", f))
```

Comment:
Its underlying type is a string, so it could be accidentally constructed by hand (`FullConversationAlias("abcd")`), or be called on an empty one (with zero value of empty string). This guards against those situations. And since it shouldn't happen if the code is written correctly, it's a `panic`